PluginTemplateProject/src/main/scala/com/alpine/templates/db/TransformationTemplate.scala

```diff
@@ -0,0 +1,195 @@
+  }
+
+
+  private def updateOutputSchema(inputSchemas: Map[String, TabularSchema],
+                                 params: OperatorParameters,
+                                 operatorSchemaManager: OperatorSchemaManager): Unit = {
+
+    //todo - create an output TabularSchema "outputSchema"
+    //todo - set the output schema of the operatorSchemaManager to "outputSchema"
+    //NOTE - for now, output schema = input schema
+    if (inputSchemas.nonEmpty) {
+      val inputSchema = inputSchemas.values.iterator.next()
+      if (inputSchema.getDefinedColumns.nonEmpty) {
+        operatorSchemaManager.setOutputSchema(inputSchema)
+
+      }
+    }
+  }
+
+}
+
+/**
+ * The DBTransformationTemplateRuntime defines the runtime behavior of the operator.
+ */
+class DBTransformationTemplateRuntime extends DBRuntime[DBTable, DBTable] {
+
+  override def onExecution(context: DBExecutionContext,
+                           input: DBTable,
+                           params: OperatorParameters,
+                           listener: OperatorListener): DBTable = {
+
+    //get the basic database output parameters
+    val outputSchema = DBParameterUtils.getDBOutputSchemaParam(params)
+    val isView = DBParameterUtils.getIsViewParam(params)
+    val outputName = DBParameterUtils.getResultTableName(params)
+    val connectionInfo = context.getDBConnectionInfo
+    val overwrite = DBParameterUtils.getOverwriteParameterValue(params)
+    val fullOutputName = getQuotedSchemaTableName(outputSchema, outputName)
+    val tableOrView = if (isView) "VIEW" else "TABLE"
+
+    //check if there is a table  or with the same name as the output table and drop according to the
+    // "overwrite"
+    if (overwrite) {
+      val stmtTable = connectionInfo.connection.createStatement()
+      //First see if a table of that name exists.
+      // This will throw an exception if there is a view with the output name,
+      // we will catch the exception and delete the view in the next block of code.
+      try {
+        listener.notifyMessage("Dropping table if it exists")
+        val dropTableStatementBuilder = new StringBuilder()
+        dropTableStatementBuilder ++= "DROP TABLE IF EXISTS " + fullOutputName + " CASCADE;"
+        stmtTable.execute(dropTableStatementBuilder.toString())
+      }
+      catch {
+        case (e: Exception) => listener.notifyMessage("A view of the name " + fullOutputName + "exists")
+      }
+      finally {
+        stmtTable.close()
+      }
+      //Now see if there is a view with the output name
+      listener.notifyMessage("Dropping view if it exists")
+      val dropViewStatementBuilder = new StringBuilder()
+      dropViewStatementBuilder ++= "DROP VIEW IF EXISTS " + fullOutputName + " CASCADE;"
+      val stmtView = connectionInfo.connection.createStatement()
+      stmtView.execute(dropViewStatementBuilder.toString())
+      stmtView.close()
+    }
+
+    //todo - pull out values from custom parameters
+
```

Comment:
Not sure.  For training, this would be accompanied by explanations for all the missing pieces in the operator, so the user would have samples of parameters to copy and paste.

On the flip side, if people are just using this from scratch without any guidance/training, then it would be useful to have a parameter example in here as well.

I was trying to set this up in a way that the end user would have to delete as little as possible to create their new operator.  My first version didn't even have SQL code included.  But then I figured it would be good to have it usable even in its most basic state.

We could potentially add a parameter like "enter a number" that doesn't change the output schema or the SQL code.  That would keep everything else as simple as possible.  Thoughts?